Flower comments

CourtneyJC opened the pull request.

CourtneyJC committed:
Add method for adding plant comments to the database

Issue #3

CourtneyJC committed:
Add test for storing comment, fix bug in storePlantComment

Issue #3

Commit:
Fix bug in storePlantComment

Comments should not be stored as ObjectId

Issue #3

Commit:
Add storePlantComment test for proper input

Issue #3

CourtneyJC committed:
Add clientside ability to leave comments on plants

Currently dumb because it sends the entire comment in the URL, but it works

Issue #3

CourtneyJC committed:
Improve clientside ability to leave comments on plants

Issue #3

lverns commented on client/src/app/plants/plant.service.ts:
```diff
@@ -21,4 +21,9 @@ export class PlantService {
         return this.http.request(this.plantUrl + "/" + id + "/" + rating).map(res => res.json());
     }
+
+    //todo can I have the entire comment be in the URL?
```
Let's remove this todo since we fixed the issue

lverns commented on client/src/app/plants/plant.component.html:
```diff
@@ -11,4 +11,13 @@ <h3 >{{plant.cultivar}}</h3>
     </div>
+    <br>
+    <div *ngIf="!commented">
+        <textarea [(ngModel)]="currentQuery" id="commentBox" rows="5" wrap="on" style="width: 95%;"></textarea>
```
We should probably rename `currentQuery` to something more accurate like `currentCommentText` or just `commentText`. 

Also, do we need the `id="commentBox"` anymore?

CourtneyJC replied:
Nope, I'll remove it. 

Do we want to keep the height at 5 spaces?

CourtneyJC replied:
Thought: do we want to have a count in the Plants metadata of how many comments each flower gets? We'd just need to add another field to the metadata in the plant database, and then increment each time a comment is left, which wouldn't be too hard.

Or do we not care at this point, since we can look at the comments individually anyway?

CourtneyJC committed:
Remove unneeded code, rename textbox variable, fix comment method

If commentbox is empty and 'Leave comment' button is clicked, it will no longer send an "Undefined" comment

Issue #3

CourtneyJC commented:
Thought: do we want to have a count in the Plants metadata of how many comments each flower gets? We'd just need to add another field to the metadata in the plant database, and then increment each time a comment is left, which wouldn't be too hard.

Or do we not care at this point, since we can look at the comments individually anyway?

lverns commented:
I think that it is easier and less error prone to just filter through the `comments` collection and count the results.
